Source/WebCore/css/CSSGradientValue.cpp

```diff
@@ -95,24 +95,22 @@ template<StopPositionResolution resolution> static inline std::optional<Length>
     if (!position)
         return std::nullopt;
 
-    if constexpr (resolution == StopPositionResolution::Deprecated) {
-        if (position->isPercentage())
-            return Length(position->floatValue(CSSUnitType::CSS_PERCENTAGE), LengthType::Percent);
-        return Length(position->floatValue(CSSUnitType::CSS_NUMBER), LengthType::Fixed);
-    } else
+    if constexpr (resolution == StopPositionResolution::Deprecated)
+        return Length(position->valueDividingBy100IfPercentage<float>(state.cssToLengthConversionData()), LengthType::Fixed);
+    else
         return Style::BuilderConverter::convertLength(state, *position);
 }
 
-static inline std::variant<std::monostate, AngleRaw, PercentRaw> computeAngularStopPosition(const RefPtr<CSSPrimitiveValue>& position)
+static inline std::variant<std::monostate, AngleRaw, PercentRaw> computeAngularStopPosition(const RefPtr<CSSPrimitiveValue>& position, Style::BuilderState& state)
 {
     if (!position)
         return std::monostate { };
 
     if (position->isPercentage())
-        return { PercentRaw { position->doubleValue(CSSUnitType::CSS_PERCENTAGE) } };
+        return { PercentRaw { position->resolveAsPercentage(state.cssToLengthConversionData()) } };
 
     if (position->isAngle())
-        return { AngleRaw { position->primitiveType(), position->doubleValue() } };
+        return { AngleRaw { CSSUnitType::CSS_DEG, position->resolveAsAngle(state.cssToLengthConversionData()) } };
```

Comment:
Since `CSS_DEG` here is explicit, maybe the template parameter specifying degrees as the unit should be as well? It would be inelegant but clearer.